src/panels/config/blueprint/blueprint-generic-editor.ts

```diff
@@ -141,6 +117,85 @@ export abstract class HaBlueprintGenericEditor extends LitElement {
     `;
   }
 
+  private renderSection(sectionKey: string, section: BlueprintInputSection) {
+    const title = section?.name || sectionKey;
+    const anyRequired =
+      section.input &&
+      Object.values(section.input).some(
+        (item) => item === null || item.default === undefined
+      );
+    const expanded = !section.collapsed || anyRequired;
+
+    return html` <ha-expansion-panel
+      outlined
+      .expanded=${expanded}
+      .noCollapse=${anyRequired}
```

Comment:
Should we use an expansion panel if it can not be collapsed anyway? Shouldn't we just use a `ha-card` in that case?

Reply:
Initially when I wrote this they were all collapsible, so the removal of collapsing was a last minute add on. 

I think I was concerned that the CSS styling might not look the same if some sections were expPanels and some were ha-cards. Just looking at it it seems like the font-size for the headers might be different, and possibly other paddings and margins as well. It could maybe be fixed but I'm not convinced the styling wouldn't further diverge in the future. 

Keeping the same class I think guarantees that collapsible/noncollapsible sections always have a similar look and feel.